### ISSUE TITLE
All files opening in Word 

### ISSUE THREAD

**Issue by nanos** [maks/MGit#525]
On my phone, all files try to open in Word, when I click the edit button. 

I've even tried to remove the word app, but now they open in the play store on the word app page. 

Is there anything I can do to change this? I have Turbo Editor installed on my phone. 

Thank you! 

**Smartphone (please complete the following information):**
 - Device: Pixel 2
 - OS: Android 10
 - App Version 1.5.7

**Comment by maks** [comment]
Hmm, looks like Turbo Editor has unfortunately not been updated to work correctly on Android 10 and Content providers which is what MGit uses when you clone repos into its internal storage area.
I'll need to look at the best way to fix this issue, but in the mean time a work around would be to remove the local repo in MGit, got into MGit settings and change your root storage location to a location that is readable by all apps (eg. Downloads), clone the repo again and then you will be able to open the files in the repo directly inside Turbo Editor using its file browser.

**Comment by nanos** [comment]
Thank you very much for that response, maks!

Can you recommend another editor that would work straight from MGit?

**Comment by maks** [comment]
Unfortunately not. I did a little bit of research and testing and have been quite surprised by the sad lack of good quality text editors available, open source and proprietary that are able to handle files via a content provider correctly.
I'll have a look into what's required to get Turbo Editor working on Android 10 correctly, as I was previously testing it with MGit on Android 8.1 and there it works fine.

**Comment by nanos** [comment]
OK, thanks! 😄

**Comment by maks** [comment]
After more research, it turn out that the Turbo Editor app [is no longer maintained by its original author](https://github.com/vmihalachi/turbo-editor/pull/164#issuecomment-549746223), the last update was published in 2015 on Google Play. Luckily it recently got a big PR updating the code from another contributor so given there are really not any other alternatives available, I've created my own fork, renamed it as Viper Edit in order to be able to publish on Google Play without confusing it with original app. Its in the process of being published now, so I'll update here once its available.

I've updated it to work correctly with [MGit in Android 10 as part of this commit](https://github.com/maks/viper-edit/commit/155befe040a2d15a9eff111ff78877cadf6383d1) so  it should now allow you to open files from MGit directly into Viper Edit, though I'll have a small commit still to make to MGit to improve how file opening for viewing and editing is done.
